core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -280,7 +294,7 @@ public long nextFetchOffset() {
         try {
             // When none of the records in the cachedState are in the AVAILABLE state, findNextFetchOffset will be false
             if (!findNextFetchOffset.get()) {
```

Comment:
We might want to clean up `findNextFetchOffset` as earlier I kept as AtomicBoolean becuase it was accessed outside the lock but seems with recent refactor we changed the code a bit and we might don't want to have atomic boolean but be a bt careful. In case you can add a jira then please assign to me, I ll do that refactor.
cc: @chirag-wadhwa5